### PR TITLE
Arnej/set rawscore for locations

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe please review
@toregge FYI

i believe we can remove the old iterator afterwards.

### CODE REVIEW

File: searchlib/src/vespa/searchlib/common/locationiterators.cpp [hunk 1]
```diff
@@ -7,6 +7,106 @@
...
+
+void
+LocationIterator::doSeek(uint32_t docId)
+{
+    while (__builtin_expect(docId < getEndId(), true)) {
+        if (__builtin_expect(docId >= _numDocs, false)) {
+            break;
+        }
+        uint32_t numValues = _location.getVec()->get(docId, &_pos[0], _pos.size());
+        if (numValues > _pos.size()) {
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/common/locationiterators.cpp:60]: consider while loop for more robustness

File: searchlib/src/vespa/searchlib/common/locationiterators.cpp [hunk 2]
```diff
@@ -7,6 +7,106 @@
...
+    setAtEnd();
+}
+
+void
+LocationIterator::doUnpack(uint32_t docId)
+{
+    uint64_t sqabsdist = std::numeric_limits<uint64_t>::max();
+    int32_t docx = 0;
+    int32_t docy = 0;
+    uint32_t numValues = _location.getVec()->get(docId, &_pos[0], _pos.size());
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/common/locationiterators.cpp:85]: should probably use already fetched values here

File: searchlib/src/vespa/searchlib/common/locationiterators.cpp (hunk 1 above)
Comment (arnej27959) [reply searchlib/src/vespa/searchlib/common/locationiterators.cpp]: done.

File: searchlib/src/vespa/searchlib/common/locationiterators.cpp (hunk 2 above)
Comment (arnej27959) [reply searchlib/src/vespa/searchlib/common/locationiterators.cpp]: fixed.

**Review by havardpe — APPROVED** [review]

**Comment by geirst** [comment]
@arnej27959 Do we have sufficient testing of the location iterators (and use of raw score)?

**Pull request merged**